File: python/11.container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        l, r = 0, len(height) - 1

        while l < r:
          lh, rh = height[l], height[r]
          current_area = (r - l) * min(lh,rh)

          max_area = max(max_area, current_area)

          if lh < lr:
            l += 1
          else:
            r -= 1

        return max_area

File: go/11.container-with-most-water.go
func maxArea(height []int) int {
    var waterLevel int
    var dif int
    var i = 0 
    var j = len(height) - 1 
    
    for i < j {
        currentLevel := 0 
        
        dif = j - i 
        
        if height[i] > height[j] {
            currentLevel = height[j] * dif
            j-- 
        } else {
            currentLevel = height[i] * dif
            i++ 
        }
        
        if waterLevel < currentLevel {
            waterLevel = currentLevel
        }
    }
    
    return waterLevel
}